Sources/TSCBasic/Process.swift

```diff
@@ -834,6 +834,20 @@ extension Process {
         }
     }
 
+    /// Execute a subprocess using async await
+    ///
+    /// - Parameters:
+    ///   - arguments: The arguments for the subprocess.
+    ///   - environment: The environment to pass to subprocess. By default the current process environment
+    /// - Returns: The process result.
+    /// - Throws: Any errors that cause the execution to fail.
+    @available(macOS 12.0, *)
+    static public func popen(arguments: [String], environment: [String : String] = ProcessEnv.vars) async throws -> ProcessResult {
+        try await withUnsafeThrowingContinuation {
+            Self.popen(arguments: arguments, environment: environment, completion: $0.resume)
+        }
+    }
```

Comment:
this would work, but still involves the an underlying dispatch queue. what would a "native" async/await implementation looks like? would it be faster and/or safer without the underlying dispatch queue?

Reply:
I think we could refactor this so the use of the dispatch queue is at a different layer.